dataflow/build.gradle

```diff
@@ -54,46 +55,59 @@ createDataflowShaded('shaded')
 createDataflowShaded('nullaway')
 createDataflowShaded('errorprone')
 
-task liveVariableTest(dependsOn: [assemble, compileTestJava], group: 'Verification') {
-    description 'Test the live variable analysis test for dataflow framework.'
-    inputs.file('tests/live-variable/Expected.txt')
-    inputs.file('tests/live-variable/Test.java')
-
-    outputs.file('tests/live-variable/Out.txt')
-    outputs.file('tests/live-variable/Test.class')
-
-    delete('tests/live-variable/Out.txt')
-    delete('tests/live-variable/Test.class')
-    doLast {
-        javaexec {
-            workingDir = 'tests/live-variable'
-            if (!JavaVersion.current().java9Compatible) {
-                jvmArgs += "-Xbootclasspath/p:${configurations.javacJar.asPath}".toString()
+/**
+ * create a task to run dataflow analysis demo
+ * @param taskName specify the taskName of created task
+ * @param dirName specify the directory name of the test file
+ * @param pkgName specify the package name of the demo
+ * @param className specify the class name of the demo
+ * @return
+ */
+def testDataflowAnalysisDemo(taskName, dirName, pkgName, className) {
+    tasks.create(name: taskName, dependsOn: [assemble, compileTestJava], group: 'Verification') {
+        description "Test the ${className} analysis test for dataflow framework."
+        inputs.file("tests/${dirName}/Expected.txt")
+        inputs.file("tests/${dirName}/Test.java")
+
+        outputs.file("tests/${dirName}/Out.txt")
+        outputs.file("tests/${dirName}/Test.class")
+
+        delete("tests/${dirName}/Out.txt")
+        delete("tests/${dirName}/Test.class")
+        doLast {
+            javaexec {
+                workingDir = "tests/${dirName}"
+                if (!JavaVersion.current().java9Compatible) {
+                    jvmArgs += "-Xbootclasspath/p:${configurations.javacJar.asPath}".toString()
+                }
+                else if (JavaVersion.current() > JavaVersion.VERSION_11) {
+                    jvmArgs += [
+                        "--add-opens", "jdk.compiler/com.sun.tools.javac.code=ALL-UNNAMED",
+                        "--add-opens", "jdk.compiler/com.sun.tools.javac.comp=ALL-UNNAMED",
+                        "--add-opens", "jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED",
+                        "--add-opens", "jdk.compiler/com.sun.tools.javac.main=ALL-UNNAMED",
+                        "--add-opens", "jdk.compiler/com.sun.tools.javac.processing=ALL-UNNAMED",
+                        "--add-opens", "jdk.compiler/com.sun.tools.javac.tree=ALL-UNNAMED",
+                        "--add-opens", "jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED"
+                    ]
+                }
+
+                classpath = sourceSets.test.runtimeClasspath
+                classpath += sourceSets.test.output
+                mainClass = "${pkgName}.${className}"
```

Comment:
Is there a reason to split `pkgName` and `className`? Instead, could you just have a single class-name parameter that contains a fully-qualified name?

Reply:
Ah, you used `className` in the test description. My suggestion would just use the directory name there.

Reply:
Ok, now I use the full qualified class name.